diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2dcd85f..87cf06a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -161,6 +161,7 @@ if (WERROR)
   add_compile_options(-Werror)
 endif()
 
+add_subdirectory(config)
 add_subdirectory(packaging)
 add_subdirectory(src)
 
diff --git a/config/CMakeLists.txt b/config/CMakeLists.txt
new file mode 100644
index 0000000..8f55c6e
--- /dev/null
+++ b/config/CMakeLists.txt
@@ -0,0 +1,2 @@
+
+add_subdirectory(linux)
diff --git a/config/linux/CMakeLists.txt b/config/linux/CMakeLists.txt
new file mode 100644
index 0000000..e9183a9
--- /dev/null
+++ b/config/linux/CMakeLists.txt
@@ -0,0 +1,2 @@
+
+add_subdirectory(network)
diff --git a/config/linux/network/CMakeLists.txt b/config/linux/network/CMakeLists.txt
new file mode 100644
index 0000000..6d09032
--- /dev/null
+++ b/config/linux/network/CMakeLists.txt
@@ -0,0 +1,3 @@
+
+add_subdirectory(cloud)
+add_subdirectory(netplan)
diff --git a/config/linux/network/cloud/99-disable-network-config.cfg b/config/linux/network/cloud/99-disable-network-config.cfg
new file mode 100644
index 0000000..033e3fd
--- /dev/null
+++ b/config/linux/network/cloud/99-disable-network-config.cfg
@@ -0,0 +1 @@
+network: { config: disabled }
\ No newline at end of file
diff --git a/config/linux/network/cloud/CMakeLists.txt b/config/linux/network/cloud/CMakeLists.txt
new file mode 100644
index 0000000..671f4e1
--- /dev/null
+++ b/config/linux/network/cloud/CMakeLists.txt
@@ -0,0 +1,7 @@
+
+install(
+    FILES
+        ${CMAKE_CURRENT_LIST_DIR}/99-disable-network-config.cfg
+        DESTINATION ${CMAKE_INSTALL_SYSCONFDIR}/cloud/cloud.cfg.d
+        COMPONENT network-config
+)
diff --git a/config/linux/network/netplan/10-network-netremote-all.yaml b/config/linux/network/netplan/10-network-netremote-all.yaml
new file mode 100644
index 0000000..27a32c7
--- /dev/null
+++ b/config/linux/network/netplan/10-network-netremote-all.yaml
@@ -0,0 +1,36 @@
+#
+# This plan defines a network topology where a set of interfaces provide gateway services to wireless network clients.
+# 
+# Netplan YAML reference: https://netplan.readthedocs.io/en/stable/netplan-yaml/
+# 
+# Instructions:
+#   1. Identify all interfaces, ethernet or otherwise, that provide network services such as DHCP, DNS, etc. to wireless
+#   network clients. This will typically be, but is not limited to, a single ethernet interface.
+#   2. For each interface identified above:
+#     a. Add a network interface definition entry according to its type ('ethernets', 'vlans', etc.) which disables all
+#     IP addressing. See the netplan yaml reference URL above for complete options if necessary. 
+#     b. Add the interface name to the 'interfaces' property of the brgateway0 bridge definition to make the interface
+#     available to all bridge slaves.
+#   3. Re-generate ("render") the plan using 'sudo netplan generate'.
+#   4. Optionally validate the generated configuration using 'sudo netplan try'.
+#   5. Apply the generated configuration using 'sudo netplan apply'.
+#
+# The default template describes a plan where a single ethernet network interface 'eno2' provides all gateway services.
+#
+
+network:
+  version: 2
+  renderer: networkd
+  ethernets:
+    eno2: # ** Change this to the interface that provides gateway network services (DHCP, DNS, etc.). **
+      # Disable DHCP on all interfaces that will be part of the virtual bridge interface.
+      dhcp4: no
+      dhcp6: no
+  bridges:
+    # Define a bridge that will act as a gateway to wireless clients.
+    brgateway0:
+      # Configure the bridge to obtain its IP address using DHCP.
+      dhcp4: yes
+      # Include interfaces that provide the gateway services to wireless clients. DO NOT ADD wireless interfaces here.
+      interfaces:
+        - eno2
diff --git a/config/linux/network/netplan/CMakeLists.txt b/config/linux/network/netplan/CMakeLists.txt
new file mode 100644
index 0000000..e16814b
--- /dev/null
+++ b/config/linux/network/netplan/CMakeLists.txt
@@ -0,0 +1,7 @@
+
+install(
+    FILES
+        ${CMAKE_CURRENT_LIST_DIR}/10-network-netremote-all.yaml
+        DESTINATION ${CMAKE_INSTALL_SYSCONFDIR}/netplan
+        COMPONENT network-config
+)
diff --git a/packaging/deb/CMakeLists.txt b/packaging/deb/CMakeLists.txt
index 23c6c4e..4e78f27 100644
--- a/packaging/deb/CMakeLists.txt
+++ b/packaging/deb/CMakeLists.txt
@@ -13,6 +13,7 @@ set(CPACK_COMPONENTS_ALL
     server 
     hostapd
     dev
+    network-config
 )
 
 include(CPack)
